projects/packages/jetpack-mu-wpcom/src/features/wpcom-media/wpcom-media-url-upload-form/index.jsx

```diff
@@ -0,0 +1,109 @@
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import { useState } from 'react';
+
+import './style.scss';
+
+const WpcomMediaUrlUploadForm = ( { ajaxUrl, action, nonce, isSiteEditor } ) => {
+	const [ url, setUrl ] = useState( '' );
+
+	const [ show, setShow ] = useState( false );
+	const [ isUploading, setIsUploading ] = useState( false );
+
+	const handleUrlChange = e => {
+		setUrl( e.target.value );
+	};
+
+	const handleSubmit = async e => {
+		if ( isUploading ) {
+			return false;
+		}
+		try {
+			new URL( url ); // eslint-disable-line no-new
+		} catch {
+			return false;
+		}
+		e.preventDefault();
+
+		const formData = new FormData();
+		formData.append( 'action', action );
+		formData.append( 'url', url );
+		formData.append( '_ajax_nonce', nonce );
+
+		setIsUploading( true );
+
+		const response = await fetch( ajaxUrl, {
+			method: 'POST',
+			body: formData,
+		} );
+
+		const { success, data } = await response.json();
+
+		if ( success ) {
+			window.wp.media.model.Attachment.get( data.attachment_id ).fetch( {
+				success: function ( attachment ) {
+					if ( isSiteEditor ) {
+						const mediaLibraryTab = window.wp.media.frame.state( 'library' );
+						mediaLibraryTab.trigger( 'open' );
+
+						window.wp.media.frame.content.get().collection.add( attachment );
+
+						const selection = mediaLibraryTab.get( 'selection' );
+						selection.reset();
+						selection.add( [ attachment ] );
+					} else {
+						window.wp.media.frame.content.get().collection.add( attachment );
```

Comment:
Opening the attachment details modal updates the reference of `content` so `collection` is no longer accessible. Directly referencing `window.wp.media.frame.controller.browserView.collection.add()` seems to do the trick, something to consider.

Reply:
Sometimes I got this error:

![image](https://github.com/user-attachments/assets/db35e73f-f6eb-4af6-8ed2-31ff30e52d64)

Reply:
Thanks all for checking, looks like another session of hours debugging/exploring Core `wp` objects is incoming... 😅